.github/workflows/non-zero-realm.yml

```
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: "Non-Zero Realm"
+on:
+  schedule:
+    - cron: "0 0 * * *" # Daily at midnight
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Branch"
+        required: true
+        type: string
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+env:
+  LC_ALL: C.UTF-8
+  CGO_ENABLED: 1
+
+jobs:
+  build:
+    name: Build
+    env:
+      HIERO_MIRROR_COMMON_REALM: 1
+      HIERO_MIRROR_COMMON_SHARD: 1
+      HIERO_MIRROR_WEB3_EVM_MODULARIZEDSERVICES: "true"
+      HIERO_MIRROR_WEB3_EVM_MODULARIZEDTRAFFICPERCENT: "1.0"
+    runs-on: hiero-mirror-node-linux-large
+    strategy:
+      fail-fast: false
+      matrix:
+        project:
+          - common
+          - graphql
+          - grpc
+          - importer
+          - monitor
+          - rest
+          - rest-java
+          - rosetta
+          - web3
+    timeout-minutes: 40 # increase it from 20 minutes since some jobs often run longer on self-hosted runners
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49 # v2.12.2
+        with:
+          egress-policy: audit
+
+      - name: Checkout Code
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Checkout Code from Branch
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ github.event.inputs.branch }}
+
+      - name: Install JDK
+        uses: actions/setup-java@c5195efecf7bdfc987ee8bae7a71cb8b11521c00 # v4.7.1
+        with:
+          distribution: temurin
+          java-version: 21
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@ac638b010cf58a27ee6c972d7336334ccaf61c96 # v4
+
+      - name: Execute Gradle
+        run: ./gradlew :${{matrix.project}}:build --scan ${{ secrets.GRADLE_ARGS }}
```

We can probably omit the build scan for this.

```suggestion
        run: ./gradlew :${{matrix.project}}:build
```